GAE/src/org/akvo/flow/rest/ImageUploadRestService.java

```diff
@@ -0,0 +1,219 @@
+            String directory = props.getProperty("images");
+            S3Util.put(bucketName, directory + "/" + fileName, file.getBytes(), file.getContentType(), true);
+            return fileName;
+        } catch (Exception e) {
+            log.log(Level.SEVERE, "Error uploading image", e);
+            return null;
+        }
+    }
+
+    private void saveQuestionAnswer(Question question, SurveyInstance formInstance, String fileName) {
+        Long questionGroupId = question.getQuestionGroupId();
+        QuestionGroup group = new QuestionGroupDao().getByKey(questionGroupId);
+        boolean repeatable = group.getRepeatable();
+        if (repeatable) {
+            saveRepeatedQuestionAnswer(question, formInstance, fileName);
+        } else {
+            saveSingleQuestionAnswer(question, formInstance, fileName);
+        }
+    }
+
+    private void saveSingleQuestionAnswer(Question question, SurveyInstance formInstance, String fileName) {
+        QuestionAnswerStoreDao questionAnswerStoreDao = new QuestionAnswerStoreDao();
+        QuestionAnswerStore existingStore = questionAnswerStoreDao.getByQuestionAndSurveyInstance(question.getKey().getId(), formInstance.getKey().getId());
+        if (existingStore == null) {
+            createAndSaveQuestionAnswer(question, formInstance, fileName, questionAnswerStoreDao, 0);
+        } else {
+            existingStore.setValue(fileName);
+            questionAnswerStoreDao.save(existingStore);
+        }
+    }
+
+    //TODO: here we do not check if the image already exists
+    //we don't know how many repetitions a given question should have
+    private void saveRepeatedQuestionAnswer(Question question, SurveyInstance formInstance, String fileName) {
+        QuestionAnswerStoreDao questionAnswerStoreDao = new QuestionAnswerStoreDao();
+        int iteration = 0;
+        List<QuestionAnswerStore> existingStores = questionAnswerStoreDao.listByQuestionAndSurveyInstance(question.getKey().getId(), formInstance.getKey().getId());
+        if (existingStores != null && !existingStores.isEmpty()) {
+            List<QuestionAnswerStore> sorted = existingStores.stream().sorted(Comparator.comparing(QuestionAnswerStore::getIteration)).collect(Collectors.toList());
+            Collections.reverse(sorted);
+            iteration = sorted.get(0).getIteration();
+        }
+        createAndSaveQuestionAnswer(question, formInstance, fileName, questionAnswerStoreDao, iteration);
+    }
+
+    private void createAndSaveQuestionAnswer(Question question, SurveyInstance formInstance, String fileName, QuestionAnswerStoreDao questionAnswerStoreDao, int iteration) {
+        QuestionAnswerStore store = new QuestionAnswerStore();
+        store.setSurveyId(formInstance.getSurveyId());
+        store.setSurveyInstanceId(formInstance.getKey().getId());
+        store.setQuestionID(question.getKey().getId() + "");
+
+        //TODO: should the date be same as collection date on the form Instance??
```

Comment:
If we are able to get the metadata of when the image was taken maybe thats better.

Reply:
I will add this together with geotag extraction in second PR